src/game/Entities/Player.cpp

```diff
@@ -1609,8 +1609,20 @@ void Player::Update(const uint32 diff)
     if (IsAlive())
     {
         m_regenTimer += diff;
-        if (m_regenTimer >= REGEN_TIME_FULL)
-            RegenerateAll(m_regenTimer / 100 * 100);
+        m_powerUpdateTimer += diff;
+        if (m_regenTimer >= REGEN_TIME_PRECISE)
```

Comment:
It's still my opinion that this regen happens more frequently than twice per second, but I don't have better data to confirm, only videos from WotLK classic, which is indeed faster, but also isn't 3.3.5a, so once again: unsure.

Reply:
Technically sending `SMSG_POWER_UPDATE` is superfluous, given that the client will already have the necessary and correct data from just the `SMSG_COMPRESSED_UPDATE_OBJECT`, but in network traffic it has been observed that `SMSG_POWER_UPDATE` is sent every 2s while regenerating power